### ISSUE TITLE
Pointclouds from callback have empty PCLHeader

### ISSUE THREAD

**Issue by mmunaro** [ros-perception/pcl_conversions#4]
When using PCL 1.7 with ROS Hydro I experienced something similar to a bug with the new PCLHeader struct.

With previous ROS, I could grab Kinect pointclouds with this kind of callback (with PointCloudT being a pcl::PointCloud of templated type):

``` c++
void pointcloudT_cb(const PointCloudT::ConstPtr& callback_cloud)
{
  ROS_INFO("PointCloudT header:");
  ROS_INFO_STREAM(callback_cloud->header);
}
```

while now, if I use that code, I obtain empty headers (of type PCLHeader):

```
[ INFO] [1376399822.191852015]: PointCloudT header:
[ INFO] [1376399822.191942196]: seq: 0 stamp: 0 frame_id:
```

One option which works is instead:

``` c++
void pointcloud2_cb(const pcl::PointCloud2& callback_cloud)
{
  ROS_INFO("PointCloud2 header:");
  ROS_INFO_STREAM(callback_cloud.header);
}
```

which produces:

```
[ INFO] [1376399822.191382507]: PointCloud2 header:
[ INFO] [1376399822.191509950]: seq: 3596 stamp: 1315582148454853 frame_id: /openni_rgb_optical_frame
```

Is it possible to grab pointclouds with callbacks waiting for PointCloudT and this is a bug or is it normal that with the new changes we can only use callbacks waiting for binary messages (PCLPointCloud2 / PointCloud2)?

I report here a link to a minimal ROS package which uses these callbacks.

Both callbacks correctly output headers in ROS Fuerte while only the one using the PointCloud2 type outputs a valid header in ROS Hydro:

https://dl.dropboxusercontent.com/u/9455593/test_callback_Fuerte.zip
https://dl.dropboxusercontent.com/u/9455593/test_callback_Hydro.zip

Cheers,
Matteo

**Comment by wjwwood** [comment]
Thanks, I'll look into this today hopefully!

**Comment by wjwwood** [comment]
Looks like this one is actually in `pcl_ros` :anger:, I'm narrowing down on it though.

**Comment by wjwwood** [comment]
I think was already fixed in https://github.com/ros-perception/perception_pcl/pull/30. That fix is in version `1.1.4`, can you see what version of `pcl_ros` you are using?

I have a catkin version of the example program you submitted:

https://github.com/wjwwood/test_callback

When I run it I get:

```
...
[ INFO] [1376614635.268529000]: PointCloud2 header:
[ INFO] [1376614635.268612000]:   stamp: 1376614635.189613056
[ INFO] [1376614635.268711000]:   seq:   857
[ INFO] [1376614635.268761000]:   frame: /camera_rgb_optical_frame
[ INFO] [1376614635.276684000]: PointCloudT header:
[ INFO] [1376614635.276762000]:   stamp: 1376614635189613
[ INFO] [1376614635.276798000]:   seq:   857
[ INFO] [1376614635.276833000]:   frame: /camera_rgb_optical_frame
[ INFO] [1376614635.329889000]: PointCloud2 header:
[ INFO] [1376614635.330841000]:   stamp: 1376614635.223823872
[ INFO] [1376614635.330895000]:   seq:   858
[ INFO] [1376614635.330944000]:   frame: /camera_rgb_optical_frame
[ INFO] [1376614635.342834000]: PointCloudT header:
[ INFO] [1376614635.342911000]:   stamp: 1376614635223823
[ INFO] [1376614635.342947000]:   seq:   858
[ INFO] [1376614635.342979000]:   frame: /camera_rgb_optical_frame
...
```

**Comment by mmunaro** [comment]
Hum...this is strange, I am using pcl_ros1.1.4-0quantal-20130724-0349-+0000 on Ubuntu 13.04.

I don't know what could be different in our configurations...

**Comment by wjwwood** [comment]
@mmunaro can you try my version of your test, to make sure I didn't accidentally work around the bug?

**Comment by mmunaro** [comment]
Ok, I am going to test is right now

**Comment by mmunaro** [comment]
It works, you were right, this problem was fixed with pcl_ros 1.1.4 but I haven't re-built my package after updating it two days ago. :(

Sorry for that and thanks for your help!

Best,
Matteo

**Comment by wjwwood** [comment]
Great, glad to hear it is fixed.
